rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/db/RMNodeData.java

```diff
@@ -0,0 +1,266 @@
+    }
+
+    @Column(length = Integer.MAX_VALUE)
+    @Type(type = "org.hibernate.type.SerializableToBlobType", parameters = @Parameter(name = SerializableToBlobType.CLASS_NAME, value = "java.lang.Object"))
+    public Client getProvider() {
+        return provider;
+    }
+
+    public void setProvider(Client provider) {
+        this.provider = provider;
+    }
+
+    @Column(length = Integer.MAX_VALUE)
+    @Type(type = "org.hibernate.type.SerializableToBlobType", parameters = @Parameter(name = SerializableToBlobType.CLASS_NAME, value = "java.lang.Object"))
+    public Permission getUserPermission() {
+        return userPermission;
+    }
+
+    public void setUserPermission(Permission nodeAccessPermission) {
+        this.userPermission = nodeAccessPermission;
+    }
+
+    @Column(length = Integer.MAX_VALUE)
+    @Type(type = "org.hibernate.type.SerializableToBlobType", parameters = @Parameter(name = SerializableToBlobType.CLASS_NAME, value = "java.lang.Object"))
+    public NodeState getState() {
+        return state;
+    }
+
+    public void setState(NodeState state) {
+        this.state = state;
+    }
+
+    public long getStateChangeTime() {
+        return stateChangeTime;
+    }
+
+    public void setStateChangeTime(long stateChangeTime) {
+        this.stateChangeTime = stateChangeTime;
+    }
+
+    @ManyToOne(fetch = FetchType.LAZY)
+    @JoinColumn(name = "NODESOURCEDATA_NAME", nullable = false, updatable = false)
+    public NodeSourceData getNodeSource() {
+        return nodeSource;
+    }
+
+    public void setNodeSource(NodeSourceData nodeSource) {
+        this.nodeSource = nodeSource;
+    }
+
+    public String getHostname() {
+        return hostname;
+    }
+
+    public void setHostname(String hostname) {
+        this.hostname = hostname;
+    }
+
+    @Column(length = Integer.MAX_VALUE)
+    @Type(type = "org.hibernate.type.SerializableToBlobType", parameters = @Parameter(name = SerializableToBlobType.CLASS_NAME, value = "java.lang.Object"))
+    public String[] getJmxUrls() {
+        return jmxUrls;
+    }
+
+    public void setJmxUrls(String[] jmxUrls) {
+        this.jmxUrls = jmxUrls;
+    }
+
+    public String getJvmName() {
+        return jvmName;
+    }
+
+    public void setJvmName(String jvmName) {
+        this.jvmName = jvmName;
+    }
+
+    /**
+     * Say whether the current node data structure reflects a particular
+     * instance of {@link Node}.
+     * @param node the node to compare to
+     * @return true if the node data has the same name and url as the given node
+     */
+    public boolean equalsToNode(Node node) {
+        if (node == null) {
+            return false;
+        }
+        return node.getNodeInformation().getName().equals(name) && node.getNodeInformation().getURL().equals(nodeUrl);
+    }
+
+    @Override
+    public int hashCode() {
+        return name.hashCode() + nodeUrl.hashCode();
+    }
+
+    @Override
+    public boolean equals(Object obj) {
```

Comment:
Can you write a few test cases which test that hashCode and equals behave the same.

Reply:
addressed

Reply:
Added RMNodeDataTest

Reply:
👍